### ISSUE TITLE
queue_redraw triggers redraw immediately multiple times per frame without queuing

### ISSUE THREAD

**Issue by anthonyronda** [godotengine/godot#99443]
### Tested versions

Tested versions 
- Reproducible in 4.3.stable
- Not reproducible in 4.2.2.stable

### System information

Godot v4.3.stable - macOS 14.6.1 - Vulkan (Forward+) - integrated Apple M1 - Apple M1 (8 Threads)

### Issue description

Calling `queue_redraw()` somewhere in the callstack of a node script's `_draw()` function caused it to redraw continuously, hanging the game on frame 0.

Although `queue_redraw()`'s documentation says [a redraw is triggered only once per frame even when you call it multiple times](https://docs.godotengine.org/en/stable/classes/class_canvasitem.html#class-canvasitem-method-queue-redraw), that is evidently not the case in 4.3.stable.

When inserting a workaround provided in another issue where there was a queuing regression https://github.com/godotengine/godot/issues/88953#issuecomment-1968907901, the redraw is forced to wait a frame and the problem went away. 

### Steps to reproduce

Run MRP project. Test in both 4.2.2.stable and 4.3.stable to see the difference. Uncomment the workaround provided in the project.
* In 4.2.2.stable, the TextureRect starts with a green border. When you hover over it, it has a red border. When mouse_exit is triggered, you have a blue border.
* Note that in 4.3.stable, the print statement is printing the current frame `0` thousands of times. You probably will crash and not load properly.
* In the provided commented out workaround, redraw is only called after the current frame is processed, and so the game loads and runs normally.

### Minimal reproduction project (MRP)

[queue_redraw-triggers-redraw-immediately-without-waiting-1-frame.zip](https://github.com/user-attachments/files/17821695/queue_redraw-triggers-redraw-immediately-without-waiting-1-frame.zip)

**Comment by AThousandShips** [comment]
This sounds like:
* https://github.com/godotengine/godot/issues/95709
* https://github.com/godotengine/godot/issues/95810

And is solved by:
* https://github.com/godotengine/godot/pull/97328

If you are able to test please confirm, but should be that bug
